### PR TITLE
[WAL-916] Watcher implementation

### PR DESCRIPTION
**Pull request by Ifropc**
Add watcher class that allows to watch transaction status updates.
Example of usage:
```kotlin
suspend fun watchTransaction() {
  val watcher = anchor.watcher()
  val result = watcher.watchOneTransaction(getAuthToken(), "transaction id")

  do {
    val event = result.channel.receive()
    when (event) {
      is StatusChange ->
        println("Status changed to ${event.status}. Transaction: ${event.transaction}")
      is ChannelClosed -> println("Channel closed. Job is done")
      is ExceptionHandlerExit -> println("Exception handler exited the job")
    }
  } while (event !is ChannelClosed)
}
```

### CODE REVIEW

File: wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/Anchor.kt [hunk 1]
```diff
@@ -82,6 +91,25 @@ internal constructor(
     return Interactive(this, httpClient)
   }
 
+  /**
+   * Creates new transaction watcher
+   *
+   * @param pullDelay pull interval in which requests to the Anchor are being made.
```
Comment (JakeUrban) [inline wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/Anchor.kt:97]: I think you mean `poll`, not `pull`, right?

**Review by JakeUrban — APPROVED** [review]
LGTM, but I wonder if it makes sense to consolidate the polling logic in both `watchOneTransaction()` and `watchAsset()`. Seems like the logic is the same other than the request made and the criteria to evaluate if it should exit.

File: wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/Anchor.kt (hunk 1 above)
Comment (Ifropc) [reply wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/Anchor.kt:97]: Yep

**Comment by Ifropc** [comment]
>I wonder if it makes sense to consolidate the polling logic
I tried to refactor this place, but there is a variable re-assignment inside "logical" part, which complicates refactoring too much (so I decided to leave it as is)

**Pull request merged**